ProcessLib/HydroMechanics/HydroMechanicsFEM-impl.h

```diff
@@ -215,28 +214,41 @@ void HydroMechanicsLocalAssembler<ShapeFunctionDisplacement,
         auto& eps = _ip_data[ip].eps;
         auto const& sigma_eff = _ip_data[ip].sigma_eff;
 
-        auto const mu = gas_phase.property(MPL::PropertyType::viscosity)
-                            .template value<double>(vars, x_position, t);
+        auto const T_ref =
+            medium->hasProperty(MPL::PropertyType::reference_temperature)
+                ? medium->property(MPL::PropertyType::reference_temperature)
+                      .template value<double>(vars, x_position, t)
+                : std::numeric_limits<double>::quiet_NaN();
 
-        double const K_over_mu =
-            _process_data.intrinsic_permeability(t, x_position)[0] / mu;
+        vars[static_cast<int>(MPL::Variable::temperature)] = T_ref;
+        vars[static_cast<int>(MPL::Variable::phase_pressure)] = N_p.dot(p);
 
-        auto const alpha =
-            solid_phase.property(MPL::PropertyType::biot_coefficient)
-                .template value<double>(vars, x_position, t);
         auto const K_S = solid_material.getBulkModulus(t, x_position);
-        auto const rho_sr = solid_phase.property(MPL::PropertyType::density)
+
+        auto const K = solid.property(MPL::PropertyType::permeability)
+                           .template value<double>(vars, x_position, t);
+        auto const alpha = solid.property(MPL::PropertyType::biot_coefficient)
+                               .template value<double>(vars, x_position, t);
+        auto const rho_sr = solid.property(MPL::PropertyType::density)
                                 .template value<double>(vars, x_position, t);
-        // TODO (FZill) get fluid properties from GPML
-        double const p_fr =
-            (_process_data.fluid_type == FluidType::Fluid_Type::IDEAL_GAS)
-                ? N_p.dot(p)
-                : std::numeric_limits<double>::quiet_NaN();
-        double const rho_fr =
-            _process_data.getFluidDensity(t, x_position, p_fr);
-        double const beta_p = _process_data.getFluidCompressibility(p_fr);
-        auto const porosity = solid_phase.property(MPL::PropertyType::porosity)
+        auto const porosity = solid.property(MPL::PropertyType::porosity)
                                   .template value<double>(vars, x_position, t);
+
+        auto const mu = gas.property(MPL::PropertyType::viscosity)
+                            .template value<double>(vars, x_position, t);
+        auto const rho_fr = gas.property(MPL::PropertyType::density)
+                                .template value<double>(vars, x_position, t);
+        auto const beta_p = gas.property(MPL::PropertyType::density)
+                .template dValue<double>(vars, MPL::Variable::phase_pressure,
+                                         x_position, t) / rho_fr;
+
+        auto const storage = medium->hasProperty(MPL::PropertyType::storage)
```

Comment:
This conditional means that you can either use a constant _storage_ parameter (combined fluid and solid compressibilities along with some poro-mechanical magic and porosity) OR to use the concept of compressibilities in the first place. This is OK, but I would highly recommend to get rid of the _storage_ parameter at all! Actually, there is no cost in doing so, since all of the parameters _storage_ is made of have to be provided anyway: fluid compressibility is given via the derivative of the density parameter, porosity and biot-coefficient are required, and solid phase compressibility can be taken from Youngs modulus (or defined directly). I see no benefit using a composite _storage_ parameter; otherwise, using _storage_ limits you in your choice of your other properties (you would have to reevaluate the value of _storage_ when changing porosity, biot-coeffient, or even the ambient temperature).
Please correct me if I'm wrong or if I missed something.

Reply:
No you're right. we have just one test which used the specific storage until I replaced it with the corresponding compressibility but there was somebody which wanted the possibility back to give the storage as a user input and I thought I could now do exactly this. But you're right, it is not beneficial. I will revert this.

Reply:
storage term is now calculated as before.
And since the compressibility is now fixed as drho_dp the benchmark InjectionProduction1D was inccompatible with the current implementation. I could recreate the same storage with the correct rho_fr = f(p) but since rho_fr is then not constant it influences the results. Thus I updated the solution files of this test. Since it mainly tests the equality of the monolithic and staggered theme solutions this shouldn't matter much.